classes/Search.php

```diff
@@ -229,22 +239,38 @@ public static function find(
             return false;
         }
 
-        $intersect_array = array();
-        $score_array = array();
+        $eligible_products2 = [];
+        $score_array = [];
+        $fuzzyLoop = 0;
         $words = Search::extractKeyWords($expr, $id_lang, false, $context->language->iso_code);
+        $fuzzyMaxLoop = (int) Configuration::get('PS_SEARCH_FUZZY_MAX_LOOP');
+        $psFuzzySearch = (int) Configuration::get('PS_SEARCH_FUZZY');
+        $psSearchMinWordLength = (int) Configuration::get('PS_SEARCH_MINWORDLEN');
 
         foreach ($words as $key => $word) {
-            if (!empty($word) && strlen($word) >= (int) Configuration::get('PS_SEARCH_MINWORDLEN')) {
+            if (!empty($word) && strlen($word) >= $psSearchMinWordLength) {
                 $sql_param_search = self::getSearchParamFromWord($word);
-
-                $intersect_array[] = 'SELECT DISTINCT si.id_product
-					FROM ' . _DB_PREFIX_ . 'search_word sw
-					LEFT JOIN ' . _DB_PREFIX_ . 'search_index si ON sw.id_word = si.id_word
-					WHERE sw.id_lang = ' . (int) $id_lang . '
-						AND sw.id_shop = ' . $context->shop->id . '
-						AND sw.word LIKE
-					\'' . $sql_param_search . '\'';
-
+                $sql = 'SELECT DISTINCT si.id_product
+                            FROM ' . _DB_PREFIX_ . 'search_word sw
+                            LEFT JOIN ' . _DB_PREFIX_ . 'search_index si ON sw.id_word = si.id_word
+                            WHERE sw.id_lang = ' . (int) $id_lang . '
+                                AND sw.id_shop = ' . $context->shop->id . '
+                                AND sw.word LIKE';
+
+                while (!($result = $db->executeS($sql . "'" . $sql_param_search . "';", true, false))) {
+                    if (!$psFuzzySearch ||
+                        $fuzzyLoop++ > $fuzzyMaxLoop ||
+                        !($sql_param_search = self::findClosestWeightestWord($context, $word))) {
```

Comment:
Could you use static instead of self because of legacy class :sweat_smile:

Reply:
I think it's ok.

Reply:
I didn't set any visibility for constantes. Should have ?

Reply:
It's about `self::findClosestWeightestWord`, => ̀static::findClosestWeightestWord`

`self` refers to the same class in which the `new` keyword is actually written.
`static` refers to whatever class in the hierarchy you called the method on

Reply:
and also every const / static calls :sweat_smile:

Reply:
Ok, I think I get it. 
As you can guess, I'm not a master for this side of code specifications. Thx for the training ! 
Done.